lib/ansible/modules/tacp/tacp_info.py

```diff
@@ -0,0 +1,221 @@
+    type: str
+    returned: always
+'''
+
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = {
+        "api_key": {"type": 'str', "required": True},
+        "resource": {"type": 'str', "required": True,
+                     "choices": [
+                         "application",
+                         "application_group",
+                         "category",
+                         "datacenter",
+                         "firewall_profile",
+                         "instance",
+                         "marketplace_template",
+                         "migration_zone",
+                         "site",
+                         "storage_pool",
+                         "tag",
+                         "template",
+                         "user",
+                         "vlan",
+                         "vnet"]}
+    }
+
+    result = dict(
+        changed=False,
+        args=[]
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if module.check_mode:
+        module.exit_json(**result)
+
+    result['args'] = module.params
+
+    # Define configuration
+    configuration = tacp.Configuration()
+    configuration.host = "https://manage.cp.lenovo.com"
+    configuration.api_key_prefix['Authorization'] = 'Bearer'
+    configuration.api_key['Authorization'] = module.params['api_key']
+    api_client = tacp.ApiClient(configuration)
+
+    resource_dict = {"application": tacp_utils.ApplicationResource,
+                     "application_group": tacp_utils.ApplicationGroupResource,
+                     "category": tacp_utils.CategoryResource,
+                     "datacenter": tacp_utils.DatacenterResource,
+                     "firewall_profile": tacp_utils.FirewallProfileResource,
+                     "instance": tacp_utils.ApplicationResource,
+                     "marketplace_template": tacp_utils.MarketplaceTemplateResource,  # noqa
+                     "migration_zone": tacp_utils.MigrationZoneResource,
+                     "site": tacp_utils.SiteResource,
+                     "storage_pool": tacp_utils.StoragePoolResource,
+                     "tag": tacp_utils.TagResource,
+                     "template": tacp_utils.TemplateResource,
+                     "user": tacp_utils.UserResource,
+                     "vlan": tacp_utils.VlanResource,
+                     "vnet": tacp_utils.VnetResource
+                     }
+
+    resource = resource_dict[module.params['resource']](api_client)
+
+    items = [item.to_dict() for item in resource.filter()]
+
+    if module.params['resource'] == 'migration_zone':
+        application_resource = tacp_utils.ApplicationResource(api_client)
+        category_resource = tacp_utils.CategoryResource(api_client)
+        datacenter_resource = tacp_utils.DatacenterResource(api_client)
+        for item in items:
+            if 'applications' in item:
```

Comment:
Not sure how the api is behaving, but I assume if the `item` does not have any `applications` an empty list will be there, meaning the below filter hit will take place, but won't be used. Let's properly use the dict's api and change the condition to

`if item.get('applications'):`

Reply:
yeah this is definitely better, done